xchange-gdax/src/main/java/info/bitrich/xchangestream/gdax/GDAXStreamingService.java

```diff
@@ -25,11 +30,13 @@
     private static final String SHARE_CHANNEL_NAME = "ALL";
     private final Map<String, Observable<JsonNode>> subscriptions = new HashMap<>();
     private ProductSubscription product = null;
+    GDAXStreamingExchange exchange;
 
     private WebSocketClientHandler.WebSocketMessageHandler channelInactiveHandler = null;
 
-    public GDAXStreamingService(String apiUrl) {
+    public GDAXStreamingService(String apiUrl, GDAXStreamingExchange exchange) {
```

Comment:
I am not a big fan of cyclic class dependencies like this. There should contract "parent" <- "child"  in this scenario. So if `StreamingService` needs something in particular, give it to it. But don't give it main object that holds everything - that leads to spaghetti.